### PR TITLE
Hash v1 beta

### PR DESCRIPTION
**Pull request by netcode**
HASH V1

### CODE REVIEW

File: cli/generate.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    // console.log("PTR", pathToRoute)
+    
+    if(schema.type == 'object'){
+        let tmpRes = {};
+        //build properties
+        for (const propertyKey in schema.properties) {
+            const propertyVal = schema.properties[propertyKey];
+            if(propertyVal.type == 'object'){
+                tmpRes[propertyKey] = buildResponse(propertyVal, pathToRoute)
+            }else{
```
Comment (christophetd) [inline cli/generate.js:162]: nit: could use some formatting

File: docs/config.md [hunk 2]
```diff
@@ -1,21 +1,17 @@
 # HASH Config
-You can have multiple applications with different configurations. Each application have its own folder using this structure
+Typical honeypot profile follow the this structure
```
Comment (christophetd) [inline docs/config.md:2]:
```suggestion
A typical honeypot profile follows the file structure below:
```

File: libs/log.js [hunk 3]
```diff
@@ -1,69 +1,87 @@
...
+                    winston.format.simple()
+                ),
+            });
+        },
+        file: () => {
+            if (!config.options.log_file) {
+                config.options.log_file = path.resolve(process.cwd(), 'hash.log')
+            }
+            return new winston.transports.File({
+                tailable:true,
```
Comment (christophetd) [inline libs/log.js:24]: nit: could use some formatting on all JS files

File: package.json [hunk 4]
```diff
@@ -1,36 +1,42 @@
...
-    "author": "",
-    "license": "ISC",
+    "author": "Datadog Security Research team",
+    "license": "Apache-2",
     "dependencies": {
         "@faker-js/faker": "^7.6.0",
         "chalk": "^4.1.2",
         "clear": "^0.1.0",
         "clui": "^0.3.6",
+        "commander": "^11.0.0",
```
Comment (christophetd) [inline package.json:22]: looks like this makes https://github.com/DataDog/HASH/blob/main/LICENSE-3rdparty.csv outdated, can we autogenerate it?

**Review by christophetd — COMMENTED** [review]
Left some comments - looks great overall! Two things that would be great to clarify:

- How does publication to npm happen?
- What are the default templates available? (Sprint Boot etc)

File: cli/run.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    //overwrite express error handler
+    http.use((err, req, res, next) => {
+        app.logger.error('HTTP -> 500 error: ' + err.message, { stack: err.stack });
+        res.status(200).send('!!');
+    });
+
+    //default endpoint
+    http.get('/', (req, res) => {
+        res.send('Hello, World');
```
[the remark this answers is not in the document]
Comment (netcode) [reply cli/run.js:53]: Yes that right.

File: package.json (hunk 4 above)
Comment (netcode) [reply package.json:22]: Updated it 👍🏻

**Comment by netcode** [comment]
> * How does publication to npm happen?
I got the NPM access, I now can publish to the Datadog NPM account

> * What are the default templates available? (Sprint Boot etc)
Good point, will add a new command to list the available templates and will update the document, but in another PR to not block this PR.

**Comment by christophetd** [comment]
> I got the NPM access, I now can publish to the Datadog NPM account

would be good to automate this through a GH Actions when a new tag is published

**Review by christophetd — APPROVED** [review]

**Pull request merged**